caffe2/python/pybind_state_dlpack.h

```diff
@@ -110,14 +110,16 @@ class DLPackWrapper {
     tensor->Resize(dims);
     const auto& meta = DLTypeToCaffe(dlTensor->dtype);
     tensor->ShareExternalPointer(
-        ((int8_t*)dlTensor->data) + dlTensor->byte_offset,
+        at::DataPtr(
+            (void*)(((int8_t*)dlTensor->data) + dlTensor->byte_offset),
+            static_cast<void*>(&dlMTensor),
```

Comment:
Don't take the address here. Also, the type is wrong: what you are passing here is a `DLManagedTensor**` but in the deleter you cast it to `DLManagedTensor*`